lua/orgmode/colors/markup_highlighter.lua

```diff
@@ -110,15 +114,51 @@ local function is_valid_markup_range(match, _, source, _)
 
   local start_text = get_node_text(start_node, source, -1, 1)
   local start_len = start_text:len()
-
-  local is_valid_start = (start_len < 3 or vim.tbl_contains(valid_pre_marker_chars, start_text:sub(1, 1)))
-    and start_text:sub(start_len, start_len) ~= ' '
+  is_valid_start = false
+  if start_len < 3 then
```

Comment:
If you know what this magic number stands for, please add a comment above ;)

Reply:
This is a situation where markup starts at the beginning of the line :). It definitely needs a comment.

We take markup start with offset char before and after (example: `test *bold*` would extract start as ` *b`. This allows to properly compare if before and after chars are suitable or valid for this region to be treated as valid markup.